include/sharpSAT/component_types/component_archetype.h

```diff
@@ -21,22 +21,25 @@
 
 
 #include <iostream>
+
+namespace sharpSAT {
+
 // State values for variables found during component
 // analysis (CA)
 typedef unsigned char CA_SearchState;
-#define   CA_NIL  0
-#define   CA_VAR_IN_SUP_COMP_UNSEEN  1
-#define   CA_VAR_SEEN 2
-#define   CA_VAR_IN_OTHER_COMP  4
+static const CA_SearchState CA_NIL = 0;
+static const CA_SearchState CA_VAR_IN_SUP_COMP_UNSEEN = 1;
+static const CA_SearchState CA_VAR_SEEN = 2;
+static const CA_SearchState CA_VAR_IN_OTHER_COMP = 4;
 
-#define   CA_VAR_MASK  7
+static const CA_SearchState CA_VAR_MASK = 7;
 
-#define   CA_CL_IN_SUP_COMP_UNSEEN  8
-#define   CA_CL_SEEN 16
-#define   CA_CL_IN_OTHER_COMP  32
-#define   CA_CL_ALL_LITS_ACTIVE  64
+static const CA_SearchState CA_CL_IN_SUP_COMP_UNSEEN = 8;
+static const CA_SearchState CA_CL_SEEN = 16;
+static const CA_SearchState CA_CL_IN_OTHER_COMP = 32;
+static const CA_SearchState CA_CL_ALL_LITS_ACTIVE = 64;
 
-#define   CA_CL_MASK  120
+static const CA_SearchState CA_CL_MASK = 120;
```

Comment:
Is there a reason not to put these into an enum class? Looking at the values, these are meant to be used as bit flags and using enum class we can define actual typesafe bitflags (it is possible to overload operators `|`, `&`, etc for an enum class).